[pytorch] register c10 ops for static dispatch

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34641 [pytorch] aten codegen to filter backends for default mobile build
* #35941 [pytorch] improve code analyzer to dump ops called from c++ functions
* **#35193 [pytorch] register c10 ops for static dispatch**

PR #34275 / D20266240 caused size regression.
PR #35148 / D20578316 reverted partially to fix the regression.
With buck selective build landed it should no longer cause size regression. This diff relands the reverted part of the original diff.

Differential Revision: [D20586305](https://our.internmc.facebook.com/intern/diff/D20586305/)

ljk53 committed:
[pytorch] register c10 ops for static dispatch

PR #34275 / D20266240 caused size regression.
PR #35148 / D20578316 reverted partially to fix the regression.
With buck selective build landed it should no longer cause size regression. This diff relands the reverted part of the original diff.

Differential Revision: [D20586305](https://our.internmc.facebook.com/intern/diff/D20586305/)

[ghstack-poisoned]

bhosmer reviewed (approved):
Accepting to unblock, per today's meeting

ljk53 committed:
Update on "[pytorch] register c10 ops for static dispatch"


PR #34275 / D20266240 caused size regression.
PR #35148 / D20578316 reverted partially to fix the regression.
With buck selective build landed it should no longer cause size regression. This diff relands the reverted part of the original diff.

Differential Revision: [D20586305](https://our.internmc.facebook.com/intern/diff/D20586305/)

[ghstack-poisoned]
